### PR TITLE
prevent NullReferenceExceptions on ConsumeResult

### LINKED ISSUE #919: `ConsumeResult` property exceptions when `Message` null
see #916 

when `ConsumeResult.Message` is null-  because the consume result corresponds to a `PartitionEOF` event - we should return `null` when the user attempts to access `ConsumeResult.Value` etc, not throw a null reference exception.

### PR DESCRIPTION
**Pull request by enzian**
resolves #919 by throwing a more verbose exception when accessing the
convenience properties in ConsumeResult.

### CODE REVIEW

File: src/Confluent.Kafka/Exceptions/MessageNullException.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Linq;
```
Comment (mhowlett) [inline src/Confluent.Kafka/Exceptions/MessageNullException.cs]: unneeded references

File: src/Confluent.Kafka/Exceptions/MessageNullException.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+using System.Collections.Generic;
+
+
+namespace Confluent.Kafka
+{
+    /// <summary>
+    ///     Represents an error that occures when <c>ConsumeResult</c> returns with a partition EOF signal. The message will be Null in this case and the convenience properties cannot be used.
```
Comment (mhowlett) [inline src/Confluent.Kafka/Exceptions/MessageNullException.cs]: I think too specific. Maybe: 'Thrown when there is an attempt to dereference a null Message reference.'

File: src/Confluent.Kafka/Exceptions/MessageNullException.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2018 Confluent Inc.
```
Comment (mhowlett) [inline src/Confluent.Kafka/Exceptions/MessageNullException.cs]: 2019

File: src/Confluent.Kafka/Exceptions/MessageNullException.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace Confluent.Kafka
+{
+    /// <summary>
+    ///     Represents an error that occures when <c>ConsumeResult</c> returns with a partition EOF signal. The message will be Null in this case and the convenience properties cannot be used.
+    /// </summary>
+    public class MessageNullException : NullReferenceException
+    {
+        public MessageNullException()
+            : base("Message is null")
```
Comment (mhowlett) [inline src/Confluent.Kafka/Exceptions/MessageNullException.cs]: Maybe: `Attempt to dereference null Message reference`

**Comment by mhowlett** [comment]
few comments then lgtm, thanks!

**Comment by enzian** [comment]
Finally had time to process the feedback! Thanks for the review!

**Comment by mhowlett** [comment]
... and i finally had time to look through the list of open PRs. LGTM - thanks!

**Pull request merged**